Makefile

```diff
@@ -64,11 +71,18 @@ psp_rr-objs := psp_rr_main.o psp_rr.o rr.o psp_lib.o $(lib)
 
 ext-libs := -lm -lrt -lpthread
 
+tcpdirect.o: tcpdirect.cu
+	nvcc -arch=sm_90 -O3 -g -D_GNU_SOURCE -DNO_LIBNUMA -DWITH_TCPDIRECT -c -o $@ $^
+
 tcp_rr: $(tcp_rr-objs)
 	$(CC) $(LDFLAGS) -o $@ $^ $(ext-libs)
 
 tcp_stream: $(tcp_stream-objs)
+ifdef WITH_TCPDIRECT
+	g++ $(LDFLAGS) -o $@ $^ $(ext-libs) -lc -L/usr/local/cuda/lib64 -lcudart -lcuda
```

Comment:
Curious why the $(CC) has been overriden. Is it because this is a c++ compiler? Do we have to use C++ compiler here?

Reply:
@Kaiyuan-Zhang might have a better answer, I don't remember